compose/runtime/runtime/api/desktop/runtime.api

```diff
@@ -729,10 +730,6 @@ public abstract interface class androidx/compose/runtime/State {
 	public abstract fun getValue ()Ljava/lang/Object;
 }
 
-public final class androidx/compose/runtime/SynchronizationKt {
```

Comment:
I am not sure that it is safe do this binary incompatible change (`synchronized` is `PublishedApi`, it means public). Let's add a `@Deprecated(level = DeprecationLevel.HIDDEN` to keep compatibility.

Reply:
It is almost impossible to avoid backward incompatible change here, IMHO. 

`SynchronizedObject` is erased to `java.lang.Object` for JVM platforms (as in AOSP).  We also have:

```
public final class androidx/compose/runtime/snapshots/SnapshotKt {
...
	public static final fun getLock ()Ljava/lang/Object;
…
}
```

`getLock` had another signature before:

```
public static final fun getLock ()Landroidx/compose/runtime/SynchronizedObject;
```

Even if we introduce fake `androidx/compose/runtime/SynchronizedObject` for just desktop we cannot introduce `getLock` that returns `androidx/compose/runtime/SynchronizedObject`, because `getLock` creates just plain `j.l.Object` in `jvmMain`.

If we introduce `androidx/compose/runtime/SynchronizedObject` in `jvmMain` (that is not erased to `j.l.Object`) then we will need to back port it to AOSP and it is hard to explain the reasoning for this (avoid backward incompatible change for desktop?)

Reply:
> avoid backward incompatible change for desktop

I traced the usage, it seems we have this chain:
```
foundation.TextFieldLayoutStateCache -> writable -> sync -> getLock
```
(and some usages of `writable` in third-party libraries)

So, we have `Foundation 1.6.11` that has `SynchronizedObject lock = getLock()` inlined.

We'll have a crash in this situation:
```
App -> Foundation 1.6.11 -> Runtime 1.6.11
App -> SomeLibrary -> Runtime 1.7
```

> introduce `androidx/compose/runtime/SynchronizedObject` in `jvmMain` (that is not erased to `j.l.Object`)

I would suggest to do that. Are there any disadvantages besides having +1 class?

> back port it to AOSP and it is hard to explain the reasoning

It is unfortunate that it will be post-factum from Android perspective, but avoiding incompatibility changes is a solid reasoning for AOSP. As an alternative, we can keep Android intact, and do changes only in `desktopMain`.

Reply:
>Are there any disadvantages besides having +1 class?

No. Just  foresee difficulties to backport to AOSP. Will see

>I would suggest to do that.

Ok, done (had to fight with Kotlin compiler to do that)